### ISSUE TITLE
`Sentry.Tunnel` doesn't include `X-Forwarded-To`

### ISSUE THREAD

**Issue by bruno-garcia** [getsentry/sentry-dotnet#1309]
### Summary
 
A high level overview of why we supply tunnelling in Sentry is described in [NextJS: Using the tunnel option](https://docs.sentry.io/platforms/javascript/guides/nextjs/troubleshooting/#using-the-tunnel-option).

I'm assuming we're doing something similar in our .NET backend... so to prevent ad blockers etc. getting in the way of sending data to Sentry, your Angular front end is sending these to your back end (which is sitting on host that's allowed by the CORS policy) and the back end then forwards those events to Sentry on behalf of the Angular SPA?

Assuming I've followed along so far, our tunnelling logic then needs to get the **_user's IP address_** (i.e. the ip:port of the machine calling the tunnelling endpoint) and use this value to set the `X-Forwarded-For` header when forwarding requests. 

### References

Similar to what's discussed here: https://forum.sentry.io/t/real-client-ip-with-sentry-nextjs-tunnel/15438/2

**Comment by bruno-garcia** [comment]
Doesn't seem to be working, maybe regressed?

Example: https://sentry.io/organizations/nugettrends/replays/?project=1266321&query=duration%3A%3E%3D5

https://github.com/dotnet/nuget-trends/blob/fc196aee7904b9747a58605b0bf37f115444d1e7/src/NuGetTrends.Web/Startup.cs#L100

**Comment by jamescrosswell** [comment]
@bruno-garcia what are some examples of events that **_should_** be showing the user's IP address but are actually showing the address of the .NET backend? Where does that information show up? So how would I know what the "definition of done" is for this issue (current behaviour vs expected)?

**Comment by jamescrosswell** [comment]
The NugetTrends dashboard in the example above shows:
![image](https://github.com/getsentry/sentry-dotnet/assets/728212/bf4eaa1c-611f-4a70-8990-753bd9969b92)

So it's showing **_Unknown User_** where it should be showing the IP address. This may be caused by the lack of an X forwarded header.

**Comment by jamescrosswell** [comment]
@bruno-garcia I'm wondering how best to reproduce this. I figured I'd try to get Nuget Trends running locally. This seems to require a Postgres DB but the latest dump isn't available at https://contrib.nugettrends.com/nuget-trends-contrib.dump (I get a connection timed out from cloudflare for that URL).

btw: I can't see anything in the [history for SentryTunnelMiddleware.cs](https://github.com/getsentry/sentry-dotnet/commits/main/src/Sentry.AspNetCore/SentryTunnelMiddleware.cs) that would have caused a regression.

**Comment by bruno-garcia** [comment]
> @bruno-garcia I'm wondering how best to reproduce this. I figured I'd try to get Nuget Trends running locally. This seems to require a Postgres DB but the latest dump isn't available at [contrib.nugettrends.com/nuget-trends-contrib.dump](https://contrib.nugettrends.com/nuget-trends-contrib.dump) (I get a connection timed out from cloudflare for that URL).

Yeah sorry, tracked here: https://github.com/dotnet/nuget-trends/issues/207 DBs are generatad but not getting served, I'll look at it

> btw: I can't see anything in the [history for SentryTunnelMiddleware.cs](https://github.com/getsentry/sentry-dotnet/commits/main/src/Sentry.AspNetCore/SentryTunnelMiddleware.cs) that would have caused a regression.

I think this never worked, it's not regressed. What changed was NuGet Trends using tunneling.

**Comment by rogerfar** [comment]
@bruno-garcia I dug into this issue a bit because I ran into the same problem.

Make sure that `UseForwardedHeaders` is set before `UseSentryTunneling`, otherwise the `context.Connection?.RemoteIpAddress` won't be replaced.

**Comment by bruno-garcia** [comment]
@jamescrosswell does @rogerfar 's comment help debug it? would love getting this fixed too

**Comment by jamescrosswell** [comment]
> @jamescrosswell does @rogerfar 's comment help debug it? would love getting this fixed too

Not sure how I missed this... massive apology for taking so long to reply!

In any event, @rogerfar 's comment probably would help resolve this _**if I could reproduce it**_. Without being able to reproduce, I could make some changes to the code and ask other people who can reproduce this if it helps, but I'm kind of stabbing in the dark.
